src/cryptonote_core/blockchain.cpp

```diff
@@ -2981,6 +2982,48 @@ bool Blockchain::check_tx_inputs(transaction& tx, tx_verification_context &tvc,
       return false;
     }
   }
+  else if (tx.version == transaction::version_3_deregister_tx)
+  {
+    // Check the inputs (votes) of the transaction have not been already been
+    // submitted to the blockchain under another transaction using a different
+    // combination of votes.
+    tx_extra_service_node_deregister deregister;
+    if (!get_service_node_deregister_from_tx_extra(tx.extra, deregister))
+    {
+      MERROR_VER("TX version deregister did not have the deregister metadata in the tx_extra");
+      return false;
+    }
+
+    const uint64_t height            = deregister.block_height;
+    const size_t num_blocks_to_check = loki::service_node_deregister::VOTE_LIFETIME_BY_HEIGHT * 2;
```

Comment:
Possibly. Votes stay around for VOTE_LIFETIME_BY_HEIGHT && VOTE_LIFETIME_BY_HEIGHT * TARGET_DIFFICULTY_V2 (so time in seconds since the vote was created), so votes may be included in blocks after the vote lifetime depending on if blocks are mined real quick. Maybe 2x the maximum is overkill.

Reply:
Remove timestamp checks, make it height only.

Add height check to check_tx_inputs, verify that check_tx_inputs only works on transactions that are being considered for the next block.

Reply:
Gonna need clarification on that height check.

Reply:
> so votes may be included in blocks after the vote lifetime depending on if blocks are mined real quick

Please make it so that votes are limited by exactly vote lifetime blocks, no more, even if blocks are mined quickly. 2h expected time is more than enough.